go/admin/server/server_test.go

```diff
@@ -152,3 +152,350 @@ func TestMultiAdminServerGetDatabase(t *testing.T) {
 		assert.Equal(t, []string{"cell1", "cell2"}, resp.Database.Cells)
 	})
 }
+
+func TestMultiAdminServerGetCellNames(t *testing.T) {
+	ctx := t.Context()
+	ts := memorytopo.NewServer(ctx)
+	logger := slog.New(slog.NewTextHandler(os.Stderr, nil))
+	server := NewMultiAdminServer(ts, logger)
+
+	t.Run("empty topology returns empty list", func(t *testing.T) {
+		req := &multiadminpb.GetCellNamesRequest{}
+		resp, err := server.GetCellNames(ctx, req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		assert.Empty(t, resp.Names)
+	})
+
+	t.Run("returns all cell names", func(t *testing.T) {
+		// Create test cells
+		cells := []*clustermetadatapb.Cell{
+			{Name: "cell1", ServerAddresses: []string{"localhost:2379"}, Root: "/multigres/cell1"},
+			{Name: "cell2", ServerAddresses: []string{"localhost:2380"}, Root: "/multigres/cell2"},
+		}
+
+		for _, cell := range cells {
+			err := ts.CreateCell(ctx, cell.Name, cell)
+			require.NoError(t, err)
+		}
+
+		req := &multiadminpb.GetCellNamesRequest{}
+		resp, err := server.GetCellNames(ctx, req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		assert.ElementsMatch(t, []string{"cell1", "cell2"}, resp.Names)
+	})
+}
+
+func TestMultiAdminServerGetDatabaseNames(t *testing.T) {
+	ctx := t.Context()
+	ts := memorytopo.NewServer(ctx)
+	logger := slog.New(slog.NewTextHandler(os.Stderr, nil))
+	server := NewMultiAdminServer(ts, logger)
+
+	t.Run("empty topology returns empty list", func(t *testing.T) {
+		req := &multiadminpb.GetDatabaseNamesRequest{}
+		resp, err := server.GetDatabaseNames(ctx, req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		assert.Empty(t, resp.Names)
+	})
+
+	t.Run("returns all database names", func(t *testing.T) {
+		// Create test databases
+		databases := []*clustermetadatapb.Database{
+			{Name: "db1", DurabilityPolicy: "none", Cells: []string{"cell1"}},
+			{Name: "db2", DurabilityPolicy: "none", Cells: []string{"cell1"}},
+		}
+
+		for _, db := range databases {
+			err := ts.CreateDatabase(ctx, db.Name, db)
+			require.NoError(t, err)
+		}
+
+		req := &multiadminpb.GetDatabaseNamesRequest{}
+		resp, err := server.GetDatabaseNames(ctx, req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		assert.ElementsMatch(t, []string{"db1", "db2"}, resp.Names)
+	})
+}
+
+func TestMultiAdminServerGetGateways(t *testing.T) {
+	ctx := t.Context()
+	ts := memorytopo.NewServer(ctx)
+	logger := slog.New(slog.NewTextHandler(os.Stderr, nil))
+	server := NewMultiAdminServer(ts, logger)
+
+	t.Run("get gateways with empty topology", func(t *testing.T) {
+		req := &multiadminpb.GetGatewaysRequest{}
+		resp, err := server.GetGateways(ctx, req)
+
+		require.NoError(t, err)
+		require.NotNil(t, resp)
+		assert.Empty(t, resp.Gateways)
+	})
+
+	t.Run("get gateways filtered by non-existent cell", func(t *testing.T) {
```

Comment:
should this error or return empty? I could go either way.

Reply:
On second thought, I'm going to collect and return any errors with partial results. That is better also for the scenario where one of the cells is unreachable from the admin server.